xnet/writer_test.go

```diff
@@ -244,6 +244,41 @@ func TestIfGetAddrsByNameReturnsErrorIfNoConsulConnection(t *testing.T) {
 	assert.Empty(t, addr)
 }
 
+func TestIfGetAddrsByNameReturnsOnlyHealthyInstancesFromConsul(t *testing.T) {
+	config, server := createTestConsulServer(t)
+	consulApiClient, err := api.NewClient(config)
+	defer stopConsul(server)
+
+	require.NoError(t, err)
+
+	// given
+	agent := consulApiClient.Agent()
+	for id, name := range []string{"A", "A"} {
+		err = agent.ServiceRegister(registration(id, name))
+		require.NoError(t, err)
+	}
+
+	server.AddCheck(t, "check", "0", "critical")
+
+	client := consulDiscoveryServiceClient{client: consulApiClient}
+
+	testCases := []struct {
+		name string
+		want []Address
+	}{
+		{"A", []Address{"192.0.2.1:1"}},
```

Comment:
ok, made it a bit simpler,

> I don't see where is the registration of a healthy check

it is by default

Reply:
Well, that's the thing. I believe you're registering a health check for something completely different.
You register an instance of service with name and ID equal to "A" and then add a health check "check" for service with ID "0". I guess if you revert your code change the test passes regardless of your change. The implicity of the default operation and lack of labels/variables of meaningful items hides what is the intention of the test. My suggestion is to create variables for `serviceName` and `serviceId` and register the check for the same variable as the service registration. And please do let me know whether the test fails when you revert the original PR change.

Reply:
The meaning of test is quite clear. Not exactly as you describe, you add {id: 0, service: "A", id: 1, service: "A"}, therefore ID is not "A".

@ojagodzinski You can add "passing" status for service with ID 1, to be more explicit. But just, cherry on top.

Reply:
I missed that part where id is equal to 0. It's a bit implicit though (I didn't pay enough attention to the two variables returned from range over the ["A", "A"] array. As a reader of the code I prefer explicit code – if I register ID "0", it would be perfect to have it in the actual text of the code so I don't have to execute the code in my head :)